sphinx/environment/adapters/toctree.py

```diff
@@ -281,9 +283,41 @@ def get_toctree_ancestors(self, docname: str) -> List[str]:
             d = parent[d]
         return ancestors
 
+    ET = TypeVar('ET', bound=Element)
+
+    def _toctree_copy(self, node: ET, depth: int, maxdepth: int, collapse: bool = False) -> ET:
+        """Utility: deep-copy a TOC but omit things with the semantics of ._toctree_prune.
+
+        !!! MUST be kept consistent with ._toctree_prune !!!
```

Comment:
What do you mean by 'consistent' here?

A

Reply:
While `_toctree_prune` deletes unneeded stuff from a (previously done) deep copy of the toctree, `_toctree_copy` does the deep copy itself but only copies the parts, that `_toctree_prune` would not delete. So assuming, that `_toctree_prune` would be adjusted in order to e.g. not delete a particular node type, then `_toctree_copy` would also need to be adjusted to copy that particular node type.

TBH we didn't fully understand the different node types involved here. It would probably be best, to completely get rid of `_toctree_prune` in order not to have this consistency problem any more (maybe it is not that big of a problem if there aren't any adjustments to that code done in the foreseeable future). We tried getting rid of `_toctree_prune` but found that it is used in another place and didn't feel confident enough to do the necessary changes.